pkg/system_api.go

```diff
@@ -174,14 +189,14 @@ type Component struct {
 	Properties       Properties           `json:"properties,omitempty" gorm:"type:properties"`
 	Components       Components           `json:"components,omitempty" gorm:"-"`
 	ParentId         *uuid.UUID           `json:"parent_id,omitempty"` //nolint
-	Selectors        v1.ResourceSelectors `json:"selectors,omitempty" gorm:"resourceSelectors"`
-	ComponentChecks  v1.ComponentChecks   `json:"-" gorm:"componentChecks"`
+	Selectors        v1.ResourceSelectors `json:"selectors,omitempty" gorm:"resourceSelectors" swaggerignore:"true"`
+	ComponentChecks  v1.ComponentChecks   `json:"-" gorm:"componentChecks" swaggerignore:"true"`
 	Checks           Checks               `json:"checks,omitempty" gorm:"-"`
-	Configs          *Configs             `json:"-" gorm:"type:configs"`
+	Configs          Configs              `json:"configs" gorm:"type:configs"`
```

Comment:
I don't think we need these configs in json, they just represent what was provided by the users